### PR TITLE
Added CD (Continuous Deployment) snap packaging support

### PR DESCRIPTION
**Pull request by deepsidhu1313**
* Added required files to create a snap package. Launchpad account will automate the build whenever code is committed to git repo

# What is a snap?

A snap :
* is a squashFS filesystem containing your app code and a snap.yaml file containing specific metadata. It has a read-only file-system and, once installed, a writable area.
* is self-contained. It bundles most of the libraries and run-times it needs and can be updated and reverted without affecting the rest of the system.
* is confined from the OS and other apps through security mechanisms, but can exchange content and functions with other snaps according to fine-grained policies controlled by the user and the OS defaults.

//TODO
* Try reducing the size of final snap packages by removing unnecessary packages from priming stage

Signed-off-by: Navdeep Singh Sidhu <deepsidhu1313@users.noreply.github.com>

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,130 @@
+          name:                   sqlitebrowser  # the name of the snap
+          # version:                3.10.1  # the version of the snap
+          version: master
+          version-script:  printf "`cat currentrelease | head -n 1 `-` git -C . rev-parse --short HEAD`"
+
+          summary:                DB browser for Sqlite  # 79 char long summary
```
Comment (justinclift) [inline snap/snapcraft.yaml]: Should probably be "DB Browser for SQLite". :smile:

**Comment by justinclift** [comment]
Good stuff @deepsidhu1313. :smile:

Apart from the one wording adjustment, this looks good to me. Should we merge this now (after the wording update), or wait until the TODO item is done?

File: snap/snapcraft.yaml (hunk 1 above)
Comment (deepsidhu1313) [reply snap/snapcraft.yaml]: Sure i will make that change.

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
-          # version:                3.10.1  # the version of the snap
-          version: master
-          version-script:  printf "`cat currentrelease | head -n 1 `-` git -C . rev-parse --short HEAD`"
+          # version:              3.10.1  # the version of the snap
+          version:                master
+          version-script:         printf "`cat currentrelease | head -n 1 `-`git rev-list master --count`-` git -C . rev-parse --short HEAD`"
 
-          summary:                DB browser for Sqlite  # 79 char long summary
-          description:            GUI editor for SQLite databases  # a longer description for the snap
+          summary:                DB browser for SQLite  # 79 char long summary
```
Comment (justinclift) [inline snap/snapcraft.yaml]: "browser" -> "Browser" :wink:

**Comment by justinclift** [comment]
With the newly added shell script code, is there a source URL for it?

Just thinking we should probably check the license of the source first, just to be on the safe side.

And no, I'm not adding roadblocks on purpose.  I'd like to see this merged too. :smile:

**Comment by deepsidhu1313** [comment]
That's ok. I am learning new things here !! :smile:

**Comment by justinclift** [comment]
Awesome.  It's [even MIT licensed](https://github.com/snapcrafters/gimp/blob/381bef98fcad70b44f63baef089492fb589cdd0d/LICENSE), so there's no worries about licensing.  That's good news.

Let me know when you want this merged, and I'll press the button. :grinning:

**Comment by deepsidhu1313** [comment]
Good to go Justin !!

**Comment by justinclift** [comment]
Done.  Thanks heaps @deepsidhu1313. :smile:

I guess we should add this to the README and main website now too?  Want to make another PR or two for those bits?

**Pull request merged**